From 7a1d30a734975e8c1f732de8ce12826afdc360e3 Mon Sep 17 00:00:00 2001
From: Yanjiang Jin <yanjiang.jin@windriver.com>
Date: Thu, 27 Jun 2013 12:43:20 +0800
Subject: [PATCH 2/2] bcm-xlp: CPU_HOTPLUG: disable IRQ_FORCED_THREADING

Disable IRQ_FORCED_THREADING if CPU_HOTPLUG is selectd, else system
can't run when only one CPU is alive.

Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 arch/mips/Kconfig  |    1 -
 kernel/irq/Kconfig |    4 +++-
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/arch/mips/Kconfig b/arch/mips/Kconfig
index 0dfc2e6..69d504e 100644
--- a/arch/mips/Kconfig
+++ b/arch/mips/Kconfig
@@ -26,7 +26,6 @@ config MIPS
 	select GENERIC_IRQ_PROBE
 	select GENERIC_IRQ_SHOW
 	select HAVE_ARCH_JUMP_LABEL
-	select IRQ_FORCED_THREADING
 	select HAVE_MEMBLOCK
 	select HAVE_MEMBLOCK_NODE_MAP
 	select ARCH_DISCARD_MEMBLOCK
diff --git a/kernel/irq/Kconfig b/kernel/irq/Kconfig
index d1a758b..3cdf330 100644
--- a/kernel/irq/Kconfig
+++ b/kernel/irq/Kconfig
@@ -68,7 +68,9 @@ config IRQ_DOMAIN_DEBUG
 
 # Support forced irq threading
 config IRQ_FORCED_THREADING
-       bool
+	bool
+	default y
+	depends on !HOTPLUG_CPU
 
 config SPARSE_IRQ
 	bool "Support sparse irq numbering" if MAY_HAVE_SPARSE_IRQ
-- 
1.7.0.4

